src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -177,24 +175,29 @@ const FiatDetailsScreen = ({ route, navigation }: Props) => {
 
   const validateInput = () => {
     setValidInputs(false)
-    const newErrorMap = new Map<number, string | undefined>()
+    const newErrorMap = new Map<string, string | undefined>()
 
-    formFields.forEach((field, index) => {
+    formFields.forEach((field) => {
       if (field.format) {
-        fieldValues.current[index] = field.format(fieldValues.current[index])
+        fieldNamesToValues.current[field.name] = field.format(
+          fieldNamesToValues.current[field.name]
+        )
       }
-      const { isValid, errorMessage } = field.validate(fieldValues.current[index]?.trim())
+      const { isValid, errorMessage } = field.validate(
+        fieldNamesToValues.current[field.name]?.trim(),
+        fieldNamesToValues.current
+      )
       if (!isValid) {
-        newErrorMap.set(index, errorMessage)
+        newErrorMap.set(field.name, errorMessage)
       }
     })
 
     setErrors(newErrorMap)
     setValidInputs(newErrorMap.size === 0)
   }
 
-  const setInputValue = (value: string, index: number) => {
-    fieldValues.current[index] = value
+  const setInputValue = (fieldName: string, value: string) => {
```

Comment:
```suggestion
  const setInputValue = ({fieldName, value}: {fieldName: string, value: string}) => {
```
this will make it harder to introduce bugs where you swap the order of the arguments by accident

Reply:
i prefer to use destructuring whenever there are 2 consecutive positional arguments with the same type, since static type checks will not prevent this kind of bug otherwise